plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -480,6 +487,120 @@ def rpm_search_dicts(self):
                 ret.append(unit_key)
         return ret
 
+    @staticmethod
+    def _check_packages(existing_packages, new_packages):
+        """
+        Check if the new packages are the same as the existing ones.
+
+        :param existing_packages: list of packages presented in the existing erratum
+        :type  existing_packages: list of dicts
+
+        :param new_packages: list of packages presented in the new erratum
+        :type  new_packages: list of dicts
+
+        :return: True, if the lists of packages are equal
+        :rtype: bool
+        """
+        if len(existing_packages) == len(new_packages):
+            existing_packages.sort(key=itemgetter('filename'))
+            new_packages.sort(key=itemgetter('filename'))
+            return all(x == y for x, y in zip(existing_packages, new_packages))
+        return False
+
+    def merge_errata(self, other):
+        """
+        Merge two errata with the same errata_id.
+
+        There are two parts:
+        - merging of the pkglists in case of the erratum with the same id in different repositories
+        - overwriting the erratum metadata based on the `updated` field
+
+        NOTE: The second part should be eliminated after we change the way erratum is stored in the
+        MongoDB.
+
+        :param other: The erratum we are combining with the existing one
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        :return: the erratum after merge
+        :rtype: pulp_rpm.plugins.db.models.Errata
+        """
+        self.merge_pkglists(other)
+        if self.update_needed(other):
+            for field_name in self.erratum_fields_update:
+                setattr(self, field_name, getattr(other, field_name))
+        return self
+
+    def update_needed(self, other):
+        """
+        Decide based on the `updated` field if the update of the existing erratum is needed.
+
+        :param other: potentially a newer version of the erratum
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        :return: True if the other erratum is newer than the existing one
+        :rtype:  bool
+        """
+        # The `updated` field is just a string in the MongoDB, so there is no strict format for this
+        # date-time field. If we are not able to parse the `updated` field either in existing
+        # erratum or in the new erratum, the metadata of existing erratum won't be updated.
+
+        err_msg = 'Fail to update the %s erratum %s: Unable to parse the `updated` field'
+        existing_err_msg = err_msg % ('existing', self.errata_id)
+        other_err_msg = err_msg % ('uploaded', self.errata_id)
+        try:
+            existing_updated_dt = util.errata_format_to_datetime(self.updated, msg=existing_err_msg)
+            new_updated_dt = util.errata_format_to_datetime(other.updated, msg=other_err_msg)
+        except ValueError as e:
+            for msg in e.args:
+                _LOGGER.warn(msg)
```

Comment:
Shouldn't this failure be recorded in the TaskStatus results output as a failure? To do so you'll have to record the error similar to how other sync errors are recorded. I forget the details. Before figuring out how to do it, we should agree on if this suggestion even makes sense.